File: 80_Problem_1.py
# code_report Solution
# https://youtu.be/yNhgEYlTTVs

    def mostCommonWord(self, p, banned):

       words = re.sub(r'[^a-zA-Z]', ' ', p).lower().split()
       m = collections.defaultdict(int)
       ans = (0, "")
       for word in words:
          if word not in banned:
                m[word] += 1
                ans = max (ans, (m[word], word))
       return ans[1]       

File: 80_Problem_1.java
// code_report Solution
// https://youtu.be/yNhgEYlTTVs

import java.util.*;
import java.util.Arrays;
 
public String mostCommonWord(String p, String[] banned) 
{
    Set<String> b = new HashSet<>(Arrays.asList (banned));
    Map<String, Integer> m = new HashMap<> ();
    String[] words = p.replaceAll("[^a-zA-z ]" , "").toLowerCase ().split ("\\s+");
    String ans  = ""; 
    Integer max = 0;

    for (String w : words) {
        if (!b.contains (w)) {
            m.put (w, m.getOrDefault (w, 0) + 1);
            if (m.get (w) > max) {
                ans = w; 
                max = m.get (w);
            }
        }
    }
    
    return ans;
}
